src/typescript/frontend/src/app/trades/route.ts

```diff
@@ -0,0 +1,43 @@
+import { fetchSwapEvents } from "@/queries/market";
+import type { NextRequest } from "next/server";
+import { stringifyJSON } from "utils";
+
+type SwapSearchParams = {
+  marketID: string | null;
+  fromMarketNonce: string | null;
+};
+
+export type ValidSwapSearchParams = {
+  marketID: string;
+  fromMarketNonce: string;
+};
+
+const isNumber = (s: string) => !isNaN(parseInt(s));
+
+const isValidSwapSearchParams = (params: SwapSearchParams): params is ValidSwapSearchParams => {
+  const { marketID, fromMarketNonce } = params;
+  // prettier-ignore
+  return (
+    marketID !== null && isNumber(marketID) &&
+    fromMarketNonce !== null && isNumber(fromMarketNonce)
+  );
+};
+
+export async function GET(request: NextRequest) {
+  const searchParams = request.nextUrl.searchParams;
+  const params: SwapSearchParams = {
+    marketID: searchParams.get("marketID"),
+    fromMarketNonce: searchParams.get("fromMarketNonce"),
+  };
+
+  if (!isValidSwapSearchParams(params)) {
+    return new Response("Invalid chat search params.", { status: 400 });
+  }
+
+  const marketID = Number(params.marketID);
+  const fromMarketNonce = Number(params.fromMarketNonce);
+
+  const res = await fetchSwapEvents({ marketID, fromMarketNonce });
```

Comment:
We should probably cache these like we do with candlestick data- although we might want to strip most of the data first (since I think we quite literally only use like 3-4 fields from this endpoint), but it's a lot better than hitting the indexer

It should also be simpler to cache these compared to candlesticks